docs/blocks/index.md

```diff
@@ -1,51 +1,182 @@
 Blocks
 ======
 
-Blocks are, obviously, essential to the Minecraft world. They make up all of the terrain, structures, and machines. Chances are if you are interested in making a mod, then you will want to add some blocks. This page will guide you through the creation of blocks, and some of the things you can do with them.
+Blocks are essential to the Minecraft world. They make up all the terrain, structures, and machines. Chances are if you are interested in making a mod, then you will want to add some blocks. This page will guide you through the creation of blocks, and some of the things you can do with them.
 
-Creating a Block
-----------------
+One Block to Rule Them All
+--------------------------
+
+Before we get started, it is important to understand that there is only ever one of each block in the game. A world consists of thousands of references to that one block in different locations. In other words, the same block is just displayed a lot of times.
+
+Due to this, a block should only ever be instantiated once, and that is during registration. Once the block is registered, you can then use the registered reference as needed. Consider this example (see the [Registration][registration] page if you do not know what you are looking at):
+
+```java
+//BLOCKS is a DeferredRegister<Block>
+public static final RegistryObject<Block> MY_BLOCK = BLOCKS.register("my_block", () -> new Block(...));
+```
+
+After registering the block, all references to the new `my_block` should use this constant. For example, if you want to check if the block at a given position is `my_block`, the code for that would look something like this:
+
+```java
+//level.getBlockState(position) returns the blockstate placed in the given level (world) at the given position
+level.getBlockState(position).is(MyBlockRegistrationClass.MY_BLOCK.get());
+```
+
+This approach also has the convenient side effect that `block1 == block2` works and should be used instead of Java's `equals` method.
```

Comment:
This is not really a side effect. Since each registered block is effectively a singleton instance, the two objects are always exactly the same. As for the `#equals` method, it can still be used, but it's pointless since we already know the objects are the same instance.